datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1327,9 +1327,9 @@ impl SubqueryAlias {
 #[derive(Clone)]
 pub struct Filter {
     /// The predicate expression, which must have Boolean type.
-    predicate: Expr,
+    pub predicate: Expr,
     /// The incoming logical plan
-    input: Arc<LogicalPlan>,
+    pub input: Arc<LogicalPlan>,
```

Comment:
I think it is ok -- it would be nice to add some comments explaining a `Filter` should not be created directly but instead use `try_new()` and that these fields are only `pub` to support pattern matching